crate/server/src/database/redis/redis_with_findex.rs

```diff
@@ -0,0 +1,601 @@
+        let permissions_db = PermissionsDB::new(findex.clone(), label).await?;
+        Ok(Self {
+            objects_db,
+            permissions_db,
+            findex,
+            findex_key,
+            _db_key,
+            label: label.to_vec(),
+        })
+    }
+}
+
+#[async_trait]
+impl Database for RedisWithFindex {
+    /// Return the filename of the database if supported
+    fn filename(&self, _group_id: u128) -> PathBuf {
+        PathBuf::from("")
+    }
+
+    /// Insert the given Object in the database.
+    ///
+    /// A new UUID will be created if none is supplier.
+    /// This method will fail if a `uid` is supplied
+    /// and an object with the same id already exists
+    async fn create(
+        &self,
+        uid: Option<String>,
+        owner: &str,
+        object: &Object,
+        tags: &HashSet<String>,
+        _params: Option<&ExtraDatabaseParams>,
+    ) -> KResult<UniqueIdentifier> {
+        // If the uid is not provided, generate a new one
+        let uid = uid.unwrap_or_else(|| Uuid::new_v4().to_string());
+        let indexed_value = IndexedValue::Location(Location::from(uid.as_bytes()));
+
+        // the database object to index and store
+        let db_object = RedisDbObject::new(
+            object.clone(),
+            owner.to_string(),
+            StateEnumeration::Active,
+            tags.clone(),
+        );
+
+        // extract the keywords
+        let keywords = db_object.keywords();
+
+        // additions to the index
+        let mut additions = HashMap::new();
+        additions.insert(indexed_value, keywords);
+
+        //upsert the index
+        self.findex
+            .upsert(
+                &self.findex_key.to_bytes(),
+                &self.label,
+                additions,
+                HashMap::new(),
+            )
+            .await?;
+
+        // upsert the object
+        self.objects_db
+            .object_upsert(
+                &uid,
+                &RedisDbObject::new(
+                    object.clone(),
+                    owner.to_string(),
+                    StateEnumeration::Active,
+                    tags.clone(),
+                ),
+            )
+            .await?;
+
+        Ok(uid)
+    }
+
+    /// Insert the provided Objects in the database in a transaction
+    ///
+    /// A new UUID will be created if none is supplier.
+    /// This method will fail if a `uid` is supplied
+    /// and an object with the same id already exists
+    async fn create_objects(
+        &self,
+        owner: &str,
+        objects: &[(Option<String>, Object, &HashSet<String>)],
+        _params: Option<&ExtraDatabaseParams>,
+    ) -> KResult<Vec<UniqueIdentifier>> {
+        // If the uid is not provided, generate a new one
+        let mut uids = vec![];
+        let mut additions = HashMap::new();
+        let mut db_objects = HashMap::new();
+        for (uid, object, tags) in objects.iter() {
```

Comment:
`objects` is already an iterable slice
```suggestion
        for (uid, object, tags) in objects {
```

Reply:
Yes. Funny that Clippy misses that.
I guess, behind the scenes, calling `iter()` is exactly what happens